### PR TITLE
.net 6

### PR DESCRIPTION
**Pull request by helto4real**
## Breaking change

This change will be breaking. Users will have to upgrade their project files to .net 6 if they are using the compiled version. 

## Proposed change

.NET 6 is released and this PR upgrades all the project files etc to the new version and use version 10 of C#.
- [x] Upgrade projects and dependencies to .net 6
- [x] Upgrade CI to use .net 6 bits and some clean-up
- [x] Fix all new warnings for .net 6 compiler
- [x] Devcontainer to .net 6

## Type of change

- [x] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs [hunk 1]
```diff
@@ -312,7 +312,7 @@ public static CancellationTokenSource GetSourceWithTimeout(int milliSeconds = 10
             params (string attribute, object value)[] attributesTuples)
         {
             var attributes = new FluentExpandoObject();
-            foreach (var (attribute, value) in attributesTuples)
+            foreach (var (attribute, value) in attributesTuples ?? Array.Empty<(string attribute, object value)>())
```
Comment (FrankBakkerNl) [inline src/Fakes/NetDaemon.Fakes/HassClientMock.cs:315]: You could make an extensionmethod 

EmptyIfNull<T>(this Ienumerabke<T> input) => input ?? Array.Empty<T>()

Comment (FrankBakkerNl) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:315]: But keep it interal to this project

Comment (helto4real) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:315]: That is a good idea actually.

**Review by FrankBakkerNl — APPROVED** [review]
Nice, quite a few warnings indeed, but overall a borring PR😜

File: src/Fakes/NetDaemon.Fakes/HassClientMock.cs (hunk 1 above)
Comment (helto4real) [reply src/Fakes/NetDaemon.Fakes/HassClientMock.cs:315]: Hmm never got that to work. I will look at this at another time.

**Pull request merged**